front-end/public/helper.js

```diff
@@ -87,8 +89,46 @@ export const updateWorkspaceToDb = (textData, workspaceID) => {
   };
 
   fetch(`${serverAddr}/workspaces/update/${workspaceID}`, putPackedData)
-    .then(response => response.json())
-    .then((data) => {
-      console.log(data);
+    .then(response => response.json());}; 
+
+export const createContextMenus = (workspaces, account) => {
+  var workspaceIds = [];
+  workspaces.map((workspace) => {
+    if (account.workspaces.indexOf(workspace._id) !== -1) {
+      workspaceIds.push(workspace._id);
+      chrome.contextMenus.create({
+        id: workspace._id,
+        title: workspace.name,
+        contexts: ["selection"],
+        parentId: "parent",
+      });
+    }
+  });
+  return workspaceIds;
+};
+
+export const createWorkspaceContextMenus = (workspaceIds, account) => {
+  // Add all the workspaces as children
+  fetch(`${serverAddr}/workspaces`)
+    .then((r) => r.text())
+    .then((workspace_result) => {
+    // Result now contains the response text, do what you want...
+      const workspaces = JSON.parse(workspace_result);
+      workspaces.map((workspace) => {
+        if (workspaceIds.indexOf(workspace._id) === -1) { 
```

Comment:
Not sure if this is needed but correct me if I'm wrong pls!

I commented this out and it still works fine. If it turns out we don't need it, we can get rid of one of `createWorkspaceContextMenus` or `createContextMenus` since they're pretty similar and just reuse the code.